### ISSUE TITLE
Support for HTTPS Custom Domain in Azure CDN Endpoint

### ISSUE THREAD

**Issue by aaronswainonset** [hashicorp/terraform-provider-azurerm#13896]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Can you please add support for HTTPS Custom Domain in Azure CDN Endpoint.

### New or Affected Resource(s)

* azurerm_cdn_endpoint_custom_domain

### Potential Terraform Configuration

```hcl
resource "azurerm_cdn_endpoint_custom_domain" "example" {
  name            = "example-domain"
  cdn_endpoint_id = azurerm_cdn_endpoint.example.id
  host_name       = "${azurerm_dns_cname_record.example.name}.${data.azurerm_dns_zone.example.name}"
  protocol = "Https"
}
```

### References

* https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cdn_endpoint_custom_domain#host_name

**Comment by kethahel99** [comment]
Can I also suggest a  a way to reference the certificate secret name and vault id from/of the azure key vault.
Similiar to how it is done on the azurerm_frontdoor_custom_https_configuration resource:

![image](https://user-images.githubusercontent.com/25621492/145328375-ade8155a-7806-477b-bdfa-68d6b3f523c0.png)

**Comment by wasfree** [comment]
This seems to be solved in https://github.com/hashicorp/terraform-provider-azurerm/pull/13283

**Comment by tombuildsstuff** [comment]
Fixed via #13283 - thanks @wasfree
